android/src/main/java/nl/kega/reactnativeemdk/BarcodeScannerManager.java

```
@@ -27,118 +27,185 @@
         return "BarcodeScannerManager";
     }
-    
+
+    /*
+     * Called when the Android activity is resumed.
+     * Source: https://facebook.github.io/react-native/docs/native-modules-android#listening-to-lifecycle-events
+     * For more information about Android activity lifecycles: https://developer.android.com/guide/components/activities/activity-lifecycle
+     */
     @Override
     public void onHostResume() {
         if (this.scannerthread != null) {
             this.scannerthread.onHostResume();
         }
     }
 
+    /*
+     * Called when the Android activity is paused.
+     * Source: https://facebook.github.io/react-native/docs/native-modules-android#listening-to-lifecycle-events
+     * For more information about Android activity lifecycles: https://developer.android.com/guide/components/activities/activity-lifecycle
+     */
     @Override
     public void onHostPause() {
         if (this.scannerthread != null) {
             this.scannerthread.onHostPause();
         }
     }
 
+    /*
+     * Called when the Android activity is destroyed.
+     * Source: https://facebook.github.io/react-native/docs/native-modules-android#listening-to-lifecycle-events
+     * For more information about Android activity lifecycles: https://developer.android.com/guide/components/activities/activity-lifecycle
+     */
     @Override
     public void onHostDestroy() {
         if (this.scannerthread != null) {
             this.scannerthread.onHostDestroy();
         }
     }
 
+    /**
+     * Releases the scanner and EMDK before the JS bundle is destroyed.
+     * This allows the scanner to be used by another application.
+     */
     @Override
     public void onCatalystInstanceDestroy() {
         if (this.scannerthread != null) {
             this.scannerthread.onCatalystInstanceDestroy();
         }
     }
 
+    /**
+     * Helper method for getting the EMDKManager object by registering this class as a listener callback.

```

I'm not a huge fan of this implementation (totally not your fault). I think it would be useful if it threw an exception in the constructor if the required manufacturer is not found. Then none of these methods would require this check. 
```
if (this.scannerthread != null)
```
As it stands if the required manufacturer is not present everything just fails silently. It might be good to call out the fact that this method has no effect on devices that aren't made by the required manufacturers. Or maybe just make that extra clear in your comment at the top i.e. instead of 
```
The OS is checked for Zebra/Motorola Enterprise features before creating the scanner thread.
```
```
The OS is checked for Zebra/Motorola Enterprise features before creating the scanner thread.
If the OS lacks these required features, the instance returned by this constructor will perform
a NOOP for all method calls.
```